### ISSUE TITLE
Open log button has ambiguous name and role

### ISSUE THREAD

**Issue by henrycatalinismith** [henrycatalinismith/zetkin#6]
## Steps to reproduce

1. Go to the call page
2. Enable your screen reader
3. Focus the open log button

## Expected outcome

It should be announced as a button element with a text label explaining its purpose. See WCAG success criteria [1.1.1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html) and [4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) for more information.

## Actual outcome

It's announced as a link and has no label.

![VoiceOver's focus indicator on the button and the VoiceOver announcement transcript saying simply "Link"](https://user-images.githubusercontent.com/566159/191568104-5ac794e5-6f5f-4b60-accc-930c4ec99f39.png)

## Recommendation

Use `<button>` rather than `<a>` for this element and add an `aria-label` that describes what will happen when it's clicked.
